### PR TITLE
Untangling: sync GSV tabs with URL

### PR DESCRIPTION
**Pull request by fushar**
Related to:

- https://github.com/Automattic/dotcom-forge/issues/6725
- https://github.com/Automattic/dotcom-forge/issues/6747
- https://github.com/Automattic/dotcom-forge/issues/6748
- https://github.com/Automattic/dotcom-forge/issues/6781
- https://github.com/Automattic/dotcom-forge/issues/6777

## Proposed Changes

This PR updates the URL with the corresponding GSV tab's URL whenever it is clicked.

## Note

Let's ignore any weird stylings and focus on functionality. We should merge this ASAP to unblock others.

## Testing Instructions

1. Go to `<Calypso Live URL>/sites?flags=layout/dotcom-nav-redesign-v2`
2. Search a site
3. Click on a site
4. Verify that the preview pane is opened
5. Try clicking the tabs around
6. Verify that the URL changes

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by davemart-in — APPROVED** [review]
Looks good in my testing! Nicely done.

**Comment by arthur791004** [comment]
I found the monitor tab has a chance to get the `null` site on one of my simple sites
1. Click the Monitoring
2. The URL becomes `/home/undefined` and then `/stats/day/null`

**Comment by fushar** [comment]
> I found the monitor tab has a chance to get the null site on one of my simple sites

Yeah, site monitoring pages won't work on Simple sites. @davemart-in is working on creating issue to skip those tabs for such sites.

**Comment by davemart-in** [comment]
@arthur791004 yeah, I just created a https://github.com/Automattic/dotcom-forge/issues/6760 for that. We need to hide the Monitoring, PHP Logs, and Server Logs tabs for non-atomic sites.

**Review by arthur791004 — APPROVED** [review]
Tested and it works correctly 🎉

**Review by eoigal — APPROVED** [review]
Wfm - nice job @fushar - would have taken me a lot longer to put this together!

File: client/hosting-overview/index.tsx [hunk 1]
```diff
@@ -26,9 +45,12 @@ export default function () {
 		'/hosting-config/activate/:site_id',
 		siteSelection,
 		navigation,
-		redirectHomeIfIneligible,
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		redirectIfCurrentUserCannot( 'manage_options' ),
```
Comment (fushar) [inline client/hosting-overview/index.tsx:50]: The original line is using the exact same method, not sure why I need to ts-ignore it now

https://github.com/Automattic/wp-calypso/blob/9794b30bbb92a2fafc08b2de08a06ff69324835e/client/my-sites/hosting/index.js#L16

**Review by DustyReagan — APPROVED** [review]
Tests well for me! Great work!

File: client/sites-dashboard-v2/site-preview-pane/constants.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
-export const DOTCOM_OVERVIEW = 'dotcom-overview';
-export const DOTCOM_MONITORING = 'dotcom-monitoring';
-export const DOTCOM_PHP_LOGS = 'dotcom-php-logs';
-export const DOTCOM_SERVER_LOGS = 'dotcom-server-logs';
-export const DOTCOM_GITHUB_DEPLOYMENTS = 'dotcom-github-deployments';
-export const DOTCOM_HOSTING_CONFIG = 'dotcom-hosting-config';
```
Comment (cleacos) [inline client/sites-dashboard-v2/site-preview-pane/constants.ts:6]: These constants define the tab IDs; they are not used to define routes.

**Comment by DustyReagan** [comment]
Chatted with @cleacos, I'm merging this as is to unblock other issues, and we can followup on further feedback.

**Review by cleacos — APPROVED** [review]
It's OK to merge it as it is. You did a good job dealing with the URL routes. My initial concern was the URL pattern. At A4A, we have a different pattern, so here, it is handled differently, and it makes sense for you.

Any improvement could be handled in follow-up PRs. I have some of them in mind, for instance, about cleaning up the sites-dashboard-v2 folder. But it's not a blocker for this PR.

I left a comment about the usage of the Preview Pane tab IDs.

LGTM! It works as expected. Next week I could push a PR with some improvements at core/foundation level.

**Comment by lupus2k** [comment]
I noticed an error in github-deployments:
`wpcom_classic_early_release` disabled

'/github-deployments/:siteSlug?flags=layout/dotcom-nav-redesign-v2`
image

'/github-deployments/:siteSlug`
image

File: client/sites-dashboard-v2/dotcom-style.scss [hunk 3]
```diff
@@ -62,6 +62,16 @@
 	}
 }
 
+.wpcom-site .is-group-sites.is-global-sidebar-collapsed {
+	.layout__content {
+		padding: 16px 16px 16px calc(var(--sidebar-width-max));
+	}
```
Comment (Aurorum) [inline client/sites-dashboard-v2/dotcom-style.scss:68]: Just a heads up that these styles are seeping, but I'm not sure for what they were originally intended. 

Screenshot 2024-04-30 at 21 39 27

Comment (davemart-in) [reply client/sites-dashboard-v2/dotcom-style.scss:68]: Thanks for the heads up @Aurorum! We'll take a look and get this fixed up!

**Pull request merged**